tests/Main.hs

```diff
@@ -144,9 +156,32 @@ case_4_4 = testCase "Case4.4" $ withSeqState 0xffffffffffffffff $ do
     genCase n | n > 1      = Case4c (SeqSet n) (genCase (n-1))
               | otherwise  = Case4b (SeqSet 0) (SeqSet 1)
 
+#if __GLASGOW_HASKELL__ >= 706
+case_4_1b = testCase "Case4.1b" $ withSeqState 0x0 $ do
+    evaluate $ rnf1 $ (Case4a :: Case4 SeqSet)
```

Comment:
A minor point, but there's quite a bit of code re-use between the `case_4*`s and the `case_4*b`s. It'd be nice to split out the values that get `rnf`'d into separate top-level bindings.